sdk/include/opentelemetry/sdk/logs/logger_provider.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include "opentelemetry/logs/logger_provider.h"
+#include "opentelemetry/logs/noop.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/sdk/common/atomic_shared_ptr.h"
+#include "opentelemetry/sdk/logs/logger.h"
+#include "opentelemetry/sdk/logs/processor.h"
+
+// Define the maximum number of loggers that are allowed to be registered to the loggerprovider.
+// TODO: Add link to logging spec once this is added to it
+#define MAX_LOGGER_COUNT 100
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+class LoggerProvider final : public opentelemetry::logs::LoggerProvider,
+                             public std::enable_shared_from_this<LoggerProvider>
+{
+public:
+  /**
+   * Initialize a new logger provider with a specified processor
+   * @param processor The processor for this logger provider. This must
+   * not be a nullptr.
+   */
+  explicit LoggerProvider(std::shared_ptr<LogProcessor> processor) noexcept;
```

Comment:
Not needed?

Reply:
Is this not the expected way to initialize a logger provider (similar to tracer provider), or do you mean this should be added as part of simple/batch processor PR instead?

Reply:
I think this API should not exist, since the same functionality can be achieved by the following code. I think we want to keep the API orthogonal (e.g. if in the future we have sampler, we don't want to introduce a new `LoggerProvider.ctor(sampler)`).
```cpp
var provider = new LoggerProvider();
provider.AddProcessor(processor);
```

Reply:
Just ensure to test all corner cases ( eg trying to log when processor is not initialized , although I see one tests for it ).

Reply:
We could do [what traces does](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/sdk/include/opentelemetry/sdk/trace/tracer_provider.h#L28) and, when a sampler is added to logging, also have it in the constructor but have a default value so it's an optional parameter?